### PR TITLE
feat: Sign up #P2022-365

### PR DESCRIPTION
**Pull request by hiimsagan**

### CODE REVIEW

File: pages/user/register.js [hunk 1]
```diff
@@ -1,7 +1,109 @@
...
   return (
-    <div >
-        user register
-    </div>
+    <Wrapper>
+      <Input
+        id="username"
+        name="username"
+        type="text"
+        label="username"
```
Comment (wlechowicz) [inline pages/user/register.js:78]: Dulplication

**Review by wlechowicz — CHANGES_REQUESTED** [review]

File: pages/user/register.js [hunk 2]
```diff
@@ -1,7 +1,109 @@
...
+  }
+  const [allValues, setAllValues] = useState(initialState);
+  const [allErrors, setAllErrors] = useState(initialState);
+
+  const erros = {
+    userError: 'User mast be at least 5 leters',
+    passwordError: 'Password mast be at least 8 leters',
+    emailError: 'Invalid email',
+    confrimEmailError: 'Emails nedd to bee the same',
+  }
```
Comment (lenasokolova) [inline pages/user/register.js:28]: there are some typos in here, pls check

File: pages/user/register.js [hunk 3]
```diff
@@ -1,7 +1,109 @@
...
+    }
+
+    if (!(allValues.email.includes('@') && allValues.email.length > 0)) {
+      validateErrors.emailError = erros.emailError
+    }
+    if (!(allValues.email === allValues.confirmEmail && allValues.confirmEmail.length > 0)) {
+      validateErrors.confrimEmailError = erros.confrimEmailError
+    }
+
+    let size = (Object.keys(validateErrors).length)
```
Comment (lenasokolova) [inline pages/user/register.js:49]: I believe it's not the best way to name the variable, maybe smth more related to the validateErrors object's name.

File: pages/user/register.js [hunk 4]
```diff
@@ -1,7 +1,109 @@
...
+    if (allValues.password.length < 8) {
+      validateErrors.passwordError = erros.passwordError
+    }
+
+    if (!(allValues.email.includes('@') && allValues.email.length > 0)) {
+      validateErrors.emailError = erros.emailError
+    }
+    if (!(allValues.email === allValues.confirmEmail && allValues.confirmEmail.length > 0)) {
+      validateErrors.confrimEmailError = erros.confrimEmailError
+    }
```
Comment (rast-mk1) [inline pages/user/register.js:47]: could use RegExp;

checking email format and email / confirmEmail values could be (doesn't have to) inside if () for email length

Comment (marcelina-hasiak) [reply pages/user/register.js:47]: Adding type='email' to your html input provide you html validation (unless you didn't turn it off in your script) and '@' is necessary to pass this validation. I assume `allValues.confirmEmail.length > 0` is equal jus to `allValues.confirmEmail`.

**Review by marcelina-hasiak — COMMENTED** [review]
Nicely done!

File: pages/user/register.js [hunk 5]
```diff
@@ -1,7 +1,109 @@
...
+        {
+          ...prevState,
+          username: validateErrors.usernameError,
+          password: validateErrors.passwordError,
+          email: validateErrors.emailError,
+          confirmEmail: validateErrors.confrimEmailError
+        }))
+
+    }
+    console.log(newUser)
```
Comment (marcelina-hasiak) [inline pages/user/register.js:69]: Line:69 Empty newUser object appears in console even after submitting an empty form (for the first time). Please check this out ;)

File: pages/user/register.js (hunk 4 above)
Comment (weronika299) [inline pages/user/register.js:47]: If the user does not fill in the field confirm email, would it be better to use a different message?

**Pull request closed without merging**